### DISCUSSION TITLE
New star doesn't work

### DISCUSSION THREAD

**Discussion by AERProductions** [Yaro2709/New-Star#19] [Q&A] [answered]
Hello, I'm trying to install this project onto my server and the install goes fine.

The problem is that there is no way to set the game location. It is set to my main domain name and I can't add /newstar so when a user clicks "register" it shows an error page (404) because it isn't including the /newstar in the URL address. I have scoured the database, the config and all of the options and see that there is no way to set this important parameter.

User clicks register and it goes to http://www.mydomainname.com/index etc etc and it needs to go to /newstar/index but I have no way to set this option anywhere as far as I can tell.

So, this project is unusable because I can't set the game address. How am I suppose to do that?

**Comment by Yaro2709** [comment]
1. Версия php?
2. Это адрес ваш? По моему нет. Поэтому вы и не можете попасть на свой сайт. 
3. index.php

**Reply by AERProductions** [reply]
Версия PHP: 7.4.8
Ох, я тупой.
Теперь я заметил строку, в которой говорится, что он работает только с версиями от 5.6 до 7.3.
Есть ли планы обновить его для работы с 7.4.8?

https://www.aerproductions.com/newstar/index.php

Полагаю, я мог бы понизить свою версию PHP.

**Reply by Yaro2709** [reply]
7.4. - тест. Но я лично ошибок не находил, поэтому можете смело переходить. Если найдете - пишите.

**Reply by AERProductions** [reply]
Ну, как я уже упоминал вначале. Проблема в том, что игра автоматически использует мое доменное имя, и я не могу указать местоположение. Поэтому, если вы перейдете по предоставленной мной ссылке и нажмете «Зарегистрироваться», откроется страница с ошибкой 404. Это можно было бы исправить, если бы я мог изменить URL-адрес с помощью / newstar, но у меня нет возможности сделать это в настройках, конфигурации или базе данных.

**Reply by Yaro2709** [reply]
![image](https://user-images.githubusercontent.com/51800453/104774609-0bcbb380-5788-11eb-8269-42381071385b.png)
У меня все работает, смотрите.

**Reply by AERProductions** [reply]
Это мой опыт:
![myexp](https://user-images.githubusercontent.com/16365415/104774876-0ac16480-573d-11eb-8301-eee4a232fc84.png)

Когда я нажимаю «зарегистрировать», он переходит в /index.php и показывает ошибку 404, потому что указывает не на тот каталог.

Конечно, я могу просто ввести правильный каталог в URL-адрес, чтобы он работал, но он не работает сам по себе (что отвратит большинство обычных пользователей)

Возможно, это связано с тем, что я являюсь хостом сервера? Но мне показалось, что это связано с тем, как проект указывал на URL. Теперь я не уверен.

**Reply by Yaro2709** [reply]
1. Проблема возможно с браузером.
2. Проблема может быть с хостом (странное имя)
В Chrome все работает.

**Reply by AERProductions** [reply]
Я использую Opera GX

Идентификация браузера
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36 OPR/72.0.3815.465

**Reply by Yaro2709** [reply]
Советую просто поменять хостинг или браузер. На этом у меня пока все.

**Reply by AERProductions** [reply]
Протестировал его в Microsoft Edge, и он также идет на 404. Я думаю, что это связано с внутренней работой, не распознающей структуру каталогов, потому что кнопка регистрации (большая синяя) направляет в неправильный каталог.

У меня не установлены Chrome или Firefox, и я не хочу их устанавливать.

Просто нужно встроить опцию для настройки целевого URL-каталога.

Это базовая опция, предлагаемая при установке на форуме и при установке браузерных игр, таких как эта.



**Обновление -
 Хорошо, я понимаю, в чем проблема. «Зарегистрироваться» под полями входа в систему указывается прямо, но я нажимаю большую синюю кнопку «Зарегистрироваться сейчас! ->», а она не указывает прямо. Так что это проблема со ссылкой внутри проекта. Рад сообщить вам об этой проблеме. Надеюсь, это поможет, я скачаю исправление, когда вы его разместите.**

**Comment by AERProductions** [comment, marked as answer]
![dns](https://user-images.githubusercontent.com/16365415/104791747-79181e00-5761-11eb-8ee7-f564bcd22f75.png)

**Reply by Yaro2709** [reply]
Я везде захожу. Я не знаю откуда у вас такая проблема.

**Reply by AERProductions** [reply]
Он находится на первой странице индекса. Я использую браузер для проверки элементов и показываю вам, что большая кнопка регистрации не совпадает со ссылкой под полем входа. Думаю, мне просто придется покопаться в файлах и попробовать исправить самому. Я, наверное, просто удалю кнопку, если не знаю, как это исправить.

**Reply by Yaro2709** [reply]
Найдите файл \styles\templates\login\page.index.default.tpl
Найдите строчку 7
Удалите: `<a class="btn btn-lg btn-primary" href="/index.php?page=register" role="button">{$LNG.buttonRegister} --></a>`
Вставьте: `<a class="btn btn-lg btn-primary" href="index.php?page=register" role="button">{$LNG.buttonRegister} --></a>`
Это должно решить проблему. Я вас изначально не понял в чем проблема :) проблема решается легко.

**Reply by AERProductions** [reply]
Спасибо! Это решило проблему. Отличная работа.
